services/ranks_service.py

```diff
@@ -22,7 +22,7 @@ async def add_rank(self, rank: int, programme: str, user_id: str = None, offer_d
 
         if rank <= programmes_helper.programmes[programme].places:
             if offer_date is None:
-                offer_date = date(2020, 4, 15)
+                offer_date = date(year, 4, 15)
```

Comment:
You may have forgotten to import the constants module here, so this line will throw an exception.

Reply:
will resolve these in a minutr